test/test_mkldnn.py

```diff
@@ -176,14 +176,36 @@ def _test_conv_base(self, dim):
                                     dilation=dilation,
                                     bias=bias,
                                     groups=groups).float()
-            mkldnn_conv = mkldnn_utils.to_mkldnn(copy.deepcopy(conv))
+            x1 = x.clone()
+            x2 = x.clone().to_mkldnn()
+            if not train:
+                mkldnn_conv = mkldnn_utils.to_mkldnn(copy.deepcopy(conv))
+            elif train and dim != 1:
+                # TODO: enable conv1d training.
+                x1.requires_grad_()
+                x2.requires_grad_()
+                mkldnn_conv = copy.deepcopy(conv)
             with torch.backends.mkldnn.flags(enabled=False):
-                y_aten = conv(x)
-            y_mkldnn = mkldnn_conv(x.to_mkldnn()).to_dense()
-            self.assertEqual(y_aten, y_mkldnn)
-
-            self._test_serialization(mkldnn_conv, (x.to_mkldnn(),))
-            self._test_tracing(mkldnn_conv, (x.to_mkldnn(),))
+                y_aten = conv(x1)
+                if train and dim != 1:
+                    loss1 = y_aten.sum()
+                    loss1.backward()
+            if not train or (train and dim != 1):
+                y_mkldnn = mkldnn_conv(x2).to_dense()
+                self.assertEqual(y_aten, y_mkldnn)
+            if not train:
+                self._test_serialization(mkldnn_conv, (x.to_mkldnn(),))
+                self._test_tracing(mkldnn_conv, (x.to_mkldnn(),))
+            elif train and dim != 1:
+                loss2 = y_mkldnn.sum()
+                loss2.backward()
+                self.assertEqual(x1.grad, x2.grad.to_dense())
```

Comment:
please also add test that x2.grad is an mkldnn tensor.

Reply:
Added.